<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>First steps in genomic data analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">David Causeur</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-user"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pencil-alt"></span>
     
    Teaching
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="teaching.html">Overview</a>
    </li>
    <li>
      <a href="firststeps.html">First steps in data analysis</a>
    </li>
    <li>
      <a href="statdecision.html">Statistics for decision making</a>
    </li>
    <li>
      <a href="genomicdata.html">First steps in genomic data analysis</a>
    </li>
    <li>
      <a href="statlearnbiodata.html">Statistical learning of biological data</a>
    </li>
    <li>
      <a href="Lifelong.html">Lifelong learning</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-calculator"></span>
     
    Research
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="research.html">Research topics</a>
    </li>
    <li>
      <a href="publications.html">Publications</a>
    </li>
    <li>
      <a href="talks.html">Talks</a>
    </li>
    <li>
      <a href="packages.html">Packages</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="mailto:david.causeur@agrocampus-ouest.fr">
    <span class="fa fa-at"></span>
     
    Contact me
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">First steps in genomic data analysis</h1>

</div>


<p><em>Course objective</em>: Introducing statistical methods for
high-throughput gene expression data analysis focusing on the search for
genes whose expression is related to variations in experimental
covariates.</p>
<div id="section" class="section level1">
<h1>2022</h1>
<p><em>Organization</em>: The course is a one-day tutorial dedicated to
the selection of genes whose expression depends on experimental
conditions using RNA-Seq data. It introduces to the R packages DESeq2
and limma.</p>
<ul>
<li><a href="./teaching/html/slides_ADG_2022_RnaSeq.html">slides</a>, <a
href="./teaching/data/DataTest_foie_15ind_4cat4ind_10742genes_countData.txt">data</a>
and <a href="./teaching/scripts/ADG_2022_RnaSeq.Rmd">Rmd file</a>;</li>
</ul>
</div>
<div id="section-1" class="section level1">
<h1>2021</h1>
<p><em>Organization</em>: The course is divided into four sessions. Each
session is made of a lecture and practical classes. In those in-class
practical classes, students are given the opportunity to tackle
problem-solving exercises, with or without the use of R.</p>
<p>The following documents are provided for each session:</p>
<ul>
<li>A video lecture: R tutorial whose slides, data and R script are
provided;<br />
</li>
<li>A PDF and/or Rmd file containing a self-directed learning
exercise.</li>
</ul>
<p>To start with Rmarkdown: <a href="https://youtu.be/eAfbICgsc_E">brief
introduction</a>.</p>
<div id="session-1" class="section level2">
<h2>Session 1</h2>
<p><em>Objective</em>: Introducing general principles of gene expression
data normalization. Normalization is a preliminary step in genomic data
analysis consisting in identifying and removing variations only due to
technological biases.</p>
<ul>
<li>Video lecture (divided in five short clips: <a
href="https://youtu.be/9m6tjSswofU">1</a>, <a
href="https://youtu.be/3xtUQo2kI88">2</a>, <a
href="https://youtu.be/FczdN6HD-oA">3</a>, <a
href="https://youtu.be/94El7wKCip0">4</a>, <a
href="https://youtu.be/oQAssYDu1nk">5</a>) with <a
href="./teaching/PDFs/ADGSession1.pdf">slides</a>, <a
href="./teaching/data/chicken.txt">data</a> and <a
href="./teaching/scripts/ADGnormalisation.R">R script</a>;<br />
</li>
<li>In-class activities: <a
href="./teaching/PDFs/ADGexercice1.pdf">exercise</a>.</li>
</ul>
</div>
<div id="session-2" class="section level2">
<h2>Session 2</h2>
<p><em>Objective</em>: Being able to use the R package <a
href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a>
to import gene expression data into a R session and control the quality
of the gene expression measurements.</p>
<ul>
<li><a href="https://youtu.be/DnpPWLUsNe0">Video lecture</a> and <a
href="./teaching/scripts/ADGlimma.R">R script</a>;<br />
</li>
<li>In-class activities: exercise (<a
href="./teaching/PDFs/ADGexercice2.pdf">PDF</a> and <a
href="./teaching/scripts/ADGexercice2.Rmd">Rmd</a>).</li>
</ul>
</div>
<div id="session-3" class="section level2">
<h2>Session 3</h2>
<p><em>Objective</em>: Controlling false positives when selecting genes
whose mean expression is significantly related to experimental
covariates.</p>
<ul>
<li>Video lecture (divided in five short clips: <a
href="https://youtu.be/ASFvJIW1064">1</a>, <a
href="https://youtu.be/Njkv_fIISB4">2</a>, <a
href="https://youtu.be/U2C5g8XyocI">3</a>, <a
href="https://youtu.be/TCqUeA5BiB8">4</a>, <a
href="https://youtu.be/-g5PT7OLaFc">5</a>) with <a
href="./teaching/PDFs/ADGSession3.pdf">slides</a>, data(<a
href="./teaching/data/ADGfoie.txt">expression data</a>, <a
href="./teaching/data/ADGexternal.txt">external data</a>, <a
href="./teaching/data/ADGcovariates.txt">experimental covariates</a>)
and R script (<a href="./teaching/scripts/ADGSession3.R">R</a>, <a
href="./teaching/scripts/ADGSession3.Rmd">Rmd</a>, <a
href="./teaching/pdfs/ADGSession3Rmd.pdf">PDF</a>);</li>
<li>In-class activities: <a
href="./teaching/PDFs/ADGexercice3.pdf">exercise</a>.</li>
</ul>
</div>
<div id="session-4" class="section level2">
<h2>Session 4</h2>
<p><em>Objective</em>: Using clustering procedures to give more insight
to the list of selected genes. This short session is dedicated to
clustering methods in order to extract groups of microarrays whose
expression profiles are similar and groups of co-expressed genes.</p>
<ul>
<li><a href="https://youtu.be/Jas1UDy3do4">Video lecture</a> with data
(<a href="./teaching/data/ADGfoiepositive.txt">expression data for
selected genes</a>, <a
href="./teaching/data/ADGfoiepositiveadjusted.txt">factor-adjusted
expression data for selected genes</a>) and <a
href="./teaching/scripts/ADGclustering.R">R script</a>.</li>
</ul>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
