---
title: "Research topics"
---

```{r load_packages, message=FALSE, warning=FALSE, include=FALSE} 
library(fontawesome)
```

My research interests are in statistical methodology for various issues motivated by biological applications. More particularly, the focus of my recent papers is on the handling of dependence in high-dimensional statistical inference.   

 `r fa("pencil-alt", fill = "forestgreen")` **Statistical genomics** 

Dependence within high-dimensional gene expression profiles generates a strong instability of gene selection in large scale significance analysis. A proper handling of this dependence by latent factor models or more general whitening techniques improves stability of multiple testing procedures and power (see for example Friguet *et al*, 2009 [JASA], Friguet and Causeur, 2010 [CSDA], Causeur *et al*, 2011 [JSS], Hornung *et al*, 2016 [BMC Bioinf.], Hornung *et al*, 2017 [Bioinf.], Hébert *et al*, 2021 [CSDA]). 

 `r fa("pencil-alt", fill = "forestgreen")` **Functional data analysis**

Functional data are discretized observations of curves. Such data are generated by various technologies such as spectroscopy or electroencephalography (EEG). More and more study designs, such as Event-Related Potentials studies in neuroscience, aim at assessing the relationship between functional data and experimental covariates. Both for signal detection (global testing) and signal identification (search for significant intervals), dependence handling strategies can be designed to be efficient for a given pattern of association signal (see for example Causeur *et al*, 2012 [BRM], Sheu *et al*, 2016 [AoAS], Causeur *et al*, 2020 [Biometrics]).   

 `r fa("pencil-alt", fill = "forestgreen")` **High-dimensional regression and classification modeling**

Both in genomic and functional data analysis, estimation of regression and classification models has to deal with a possibly strong dependence within high-dimensional profiles of explanatory variables. Whitening procedures can help stabilizing model selection methods and improve prediction performance (see for example Perthame *et al*, 2015 [StatCo], Hébert *et al*, 2021 [Under revision]).   

 `r fa("pencil-alt", fill = "forestgreen")` **Biostatistics** 

I am involved in various research projects with a diversity of partners in biology, recently for multi-omic data integration issues (see Gondret *et al*., 2017 [BMC Gen.], Désert *et al*., 2018 [BMC Gen.]), peptidomics (see Suwareh *et al*., 2021 [Food Ch.]), electromyographic data analysis (see Comfort *et al*., 2021 [PACA]), etc.
